### DISCUSSION TITLE
Out of vram with Hires active ?

### DISCUSSION THREAD

**Discussion by cgidesign-de** [vladmandic/sdnext#2108] [Q&A]
I am using:
SD.next 2023-08-30
on Windows.

Device is:
NVIDIA GeForce RTX 3090, 24 GB vram
cuda: 11.8
cudnn: 8700
driver: 531.61

When activating _"second pass"_ and using _"Hires resize"_ I always get an out of memory error.
I tried with optimizations: xformers and Scaled-Dot-Product.

Is there something I can do to solve this?

Log says:
INFO | sd_models | Loading diffuser model: F:\...\morphxl_v10.safetensors
INFO | textual_inversion | Embeddings: loaded=0 skipped=19
INFO | sd_models | Model loaded in 6.3s (load=6.3s) native=1024
INFO | sd_models | Model load finished: {'ram': {'used': 1.11, 'total': 63.78}, 'gpu': {'used': 8.08, 'total': 24.0}, 'retries': 0, 'oom': 0}
INFO | devices | Setting Torch parameters: dtype=torch.float16 vae=torch.float16 unet=torch.float16
INFO | sd_models | Loading diffuser refiner: F:\...\sd_xl_refiner_1.0.safetensors
INFO | textual_inversion | Embeddings: loaded=0 skipped=19
INFO | sd_models | Model loaded in 3.8s (load=3.8s) native=1024
INFO | sd_models | Model load finished: {'ram': {'used': 6.8, 'total': 63.78}, 'gpu': {'used': 8.08, 'total': 24.0}, 'retries': 0, 'oom': 0}
INFO | webui | Startup time: 20.4s (torch=2.8s gradio=0.4s diffusers=0.8s libraries=0.6s scripts=3.9s onchange=0.1s ui-extensions=0.9s launch=0.1s app-started=0.1s checkpoint=10.5s)
INFO | textual_inversion | Embeddings: loaded=0 skipped=19
INFO | processing_diffusers | Hires: upscaler=4x_NickelbackFS_72000_G width=2048 height=2048 images=1
INFO | sd_models | Pipeline class changed from StableDiffusionXLPipeline to StableDiffusionXLImg2ImgPipeline
---
**ERROR** | call_queue | Exception: CUDA out of memory. Tried to allocate 4.00 GiB (GPU 0; 24.00 GiB total capacity; 19.60 GiB already allocated; 2.88 GiB free; 19.81 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
---

**Comment by vladmandic** [comment]
SD-XL is memory intensive and you're trying to upscale to 2048x2048 which takes a lot of memory.
you can try some memory optimizations in settings -> diffusers, cant say more since i don't know what your settings are.

**Comment by cgidesign-de** [comment]
Thanks for your reply. Please see my current settings below:
 
![sd-settings_1](https://github.com/vladmandic/automatic/assets/115958660/6ac3a82d-3e62-4a7f-8bd2-1734c4385eb4)

![sd-settings_2](https://github.com/vladmandic/automatic/assets/115958660/bc36b7c1-4f02-4e4a-a7dc-6394af62261d)

![sd-settings_3](https://github.com/vladmandic/automatic/assets/115958660/f149cf98-396d-4527-abe9-6dece38b47e8)

**Comment by cgidesign-de** [comment]
Is there a documentation somewhere about all those options maybe? In that case I could try find a solution myself.

**Reply by vladmandic** [reply]
try enabling:
- move base model to cpu when using refiner
- move base model to cpu when using vae
- enable vae tiling

**Reply by Disty0** [reply]
There is a wiki page:
https://github.com/vladmandic/automatic/wiki/SD-XL#vram-optimization

**Comment by cgidesign-de** [comment]
I tried it with the settings but without luck.
